pkg/service/service.go

```diff
@@ -15,177 +15,227 @@ import (
 	"context"
 	"fmt"
 	"net/http"
+	"os"
 	"time"
 
 	"github.com/edgexfoundry/device-sdk-go/internal/autoevent"
+	"github.com/edgexfoundry/device-sdk-go/internal/clients"
 	"github.com/edgexfoundry/device-sdk-go/internal/common"
 	"github.com/edgexfoundry/device-sdk-go/internal/container"
 	"github.com/edgexfoundry/device-sdk-go/internal/controller"
 	dsModels "github.com/edgexfoundry/device-sdk-go/pkg/models"
+	bootstrapContainer "github.com/edgexfoundry/go-mod-bootstrap/bootstrap/container"
 	"github.com/edgexfoundry/go-mod-bootstrap/di"
-
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
 	"github.com/edgexfoundry/go-mod-core-contracts/clients/types"
 	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
+	"github.com/edgexfoundry/go-mod-registry/registry"
 	"github.com/google/uuid"
+	"github.com/gorilla/mux"
 )
 
 var (
-	svc *Service
+	sdk *DeviceServiceSDK
```

Comment:
Remove package global and put in `dic`

Reply:
https://github.com/hahattan/device-sdk-go/blob/bf385af27ee6cdebb8cb1b80710fc8902eff977f/pkg/service/service.go#L234-L239

any elegant way to handle these 2 functions if we delete the package global here ?

Reply:
Yes, but it would be a breaking change... :-( 
I guess we have to live with it until V2.